sov-modules/sov-modules-macros/src/dispatch/derive_rpc.rs

```diff
@@ -0,0 +1,285 @@
+use proc_macro2::Ident;
+use quote::{format_ident, quote, ToTokens};
+use syn::{Attribute, FnArg, ImplItem, Meta, MetaList, Path, PathSegment, Type, Signature, PatType};
+
+
+/// Retrns an attribute with the name `rpc_method` replaced with `method`, and the index
+/// into the argument array where the attribute was found.
+fn get_method_attribute(attributes: &[Attribute]) -> Option<(Attribute, usize)> {
+    for (idx, attribute) in attributes.iter().enumerate() {
+        if let Ok(Meta::List(MetaList { path, .. })) = attribute.parse_meta() {
+            if path.is_ident("rpc_method") {
+                let mut new_attr = attribute.clone();
+                let path = &mut new_attr.path;
+                path.segments.last_mut().unwrap().ident = format_ident!("method");
+                return Some((new_attr, idx));
+            }
+        }
+    }
+    None
+}
+
+fn intermediate_trait_name(ident: &Ident) -> Ident {
+    let mut ident_str = ident.to_string();
+    ident_str.push_str("Rpc");
+    format_ident!("{}", ident_str)
+}
+
+fn jsonrpsee_rpc_macro_path() -> Path {
+    let segments = vec![
+        Ident::new("jsonrpsee", proc_macro2::Span::call_site()),
+        Ident::new("proc_macros", proc_macro2::Span::call_site()),
+        Ident::new("rpc", proc_macro2::Span::call_site()),
+    ];
+
+    let path_segments = segments
+        .into_iter()
+        .map(|ident| PathSegment {
+            ident,
+            arguments: syn::PathArguments::None,
+        });
+
+        Path {
+        leading_colon: Some(syn::Token![::](proc_macro2::Span::call_site())),
+        segments: syn::punctuated::Punctuated::from_iter(path_segments),
+    }
+}
+
+fn find_working_set_argument(sig: &Signature) -> Option<usize> {
+    for (idx, input) in sig.inputs.iter().enumerate(){
+        if let FnArg::Typed(PatType { ty, .. }) = input {
+            if let syn::Type::Reference(syn::TypeReference { elem, .. }) = *ty.clone() {
+                if let syn::Type::Path(syn::TypePath { path, .. }) = elem.as_ref() {
+                    if let Some(segment) = path.segments.last() {
+                        // TODO: enforce that the working set has exactly one angle bracketed argument
+                        if segment.ident == "WorkingSet" && !segment.arguments.is_empty() {
```

Comment:
If someone uses `WorkingSet as SomeOtherType` this will break, I don't have a good solution for this, we just need to remember to write a good compile time error message